packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -211,22 +228,37 @@ contract Multisig is PluginUUPSUpgradeable, Addresslist {
         bytes calldata _metadata,
         IDAO.Action[] calldata _actions,
         bool _approveProposal,
-        bool _tryExecution
+        bool _tryExecution,
+        uint64 _startDate,
+        uint64 _endDate
     ) external returns (uint256 proposalId) {
         uint64 snapshotBlock = block.number.toUint64() - 1;
 
         if (multisigSettings.onlyListed && !isListedAtBlock(_msgSender(), snapshotBlock)) {
             revert ProposalCreationForbidden(_msgSender());
         }
 
+        if (_startDate == 0) {
+            _startDate = block.timestamp.toUint64();
```

Comment:
Would it be cheaper to store `uint64 currentTime = block.timestamp.toUint64();` once instead of converting `block.timestamp` twice?

Reply:
then we get stack too deep and this was the easy fix for it

Reply:
```solidity
        {
            uint64 currentTimeStamp64 = block.timestamp.toUint64();
            if (_startDate == 0) {
                _startDate = currentTimeStamp64;
            }

            if (_startDate < currentTimeStamp64) {
                revert InvalidStartDate({limit: currentTimeStamp64, actual: _startDate});
            }
        }
```

Reply:
Not worth the curly brackets for this.

Reply:
I don't mind not having this change